### DISCUSSION TITLE
Make default commit message the PR description or include it

### DISCUSSION THREAD

**Discussion by leighmcculloch** [community/community#4584] [Pull Requests]
It would be helpful if the default commit messages of PRs when Squash & Merged included the contents of the PR description.

Today the default commit message is a list of the titles of all the commits in a PR. When iterating on PRs it is not uncommon for the main branch to be merged in, for review feedback to be addressed, and when squashing and merging all those iterations are not particularly important. I'm using Squash & Merge because the individual commits aren't important.

Most of the time I see others and myself write well formed PR descriptions that talk about what and why a change is being made. That discussion in the PR description would be ideal to include in the commit message so when viewing commits in things like a git blame view, all the context is right there, rather than a list of commit titles.

This is somewhat related to: https://github.com/github/feedback/discussions/4433.

**Comment by gnprice** [comment]
Upvoted. Previous discussion here: https://github.com/isaacs/github/issues/467

**Reply by ghost** [reply]
I believe [this issue in isaacs](https://github.com/isaacs/github/issues/1025) would be closer, since it's about squash-merges. But also upvoted. It's tedious to replace the squash title and body with the PR title and description. It's pretty gross trying to read squash-merge messages which are just a bullet-list of all the tiny, work-in-progress, or other noisy commits to a branch like merging master into it. (And when the gross thing's the default, it's what you'll get on most of your squash-commits. :)

**Comment by SchroederSteffen** [comment]
Just noticed: This has been shipped to github.com recently! 🚀 ❤️ 
https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/
(Now I just need it in [GitHub Enterprise](https://github.com/orgs/community/discussions/30439#discussioncomment-3462042).)
